File: java/src/main/java/com/iteso/nintendo/behaviors/attack/SmallFireAttack.java
package com.iteso.nintendo.behaviors.attack;

import com.iteso.nintendo.behaviors.attack.Attack;

public class SmallFireAttack implements Attack {
    
    private int attackDamage = 5;
    private String attackName = "Small fire";

    @Override
    public String attack() {
            return "Attacking opponent with " + attackName
                    + " causing a damage of " + attackDamage;
    }

    @Override
    public void setAttackName(String newAttackName) {
        this.attackName = newAttackName;
    }

    @Override
    public String getAttackName() {
        return this.attackName;
    }

    @Override
    public void setAttackDamage(int newAttackDamage) {
        this.attackDamage = newAttackDamage;
    } 

    @Override
    public int getAttackDamage() {
        return this.attackDamage;
    }
}

File: python/src/main/python/behaviors/attack/SmallFireAttack.py
from behaviors.attack.Attack import Attack


class SmallFireAttack(Attack):
    attackName = "Small Fire"
    attackDamage = 7

    def attack(self):
        return "Attacking opponent with " + self.attackName + \
            " causing a damage of " + str(self.attackDamage)

    def setAttackDamage(self, newDmg):
        self.attackDamage = newDmg

    def getAttackDamage(self):
        return self.attackDamage

    def setAttackName(self, newAttackName):
        self.attackName = newAttackName

    def getAttackName(self):
        return self.attackName
